### DISCUSSION TITLE
Windows port shenanigans

### DISCUSSION THREAD

**Discussion by sykhro** [zeek/spicy#904] [Show and tell]
I've been trying to get Spicy to work under Windows, in order to be able to generate parsers for an application that currently uses a lot of (buggy) hand-rolled ones.

Problems with a couple dependencies have been fixed independently. Next up would be some stuff in the `scripts` directory. I have started by porting [autogen-version](https://github.com/sykhro/spicy/blob/main/scripts/autogen-version) to python (given that there's other python scripts I figured this would've been better compared to a dedicated solution). Unfortunately shebangs are not a thing on Windows so invocation would have to be fixed on the CMake side.
Would this porting effort be welcome or might it be better to keep it as a separate patchset?

**Comment by rsmmr** [comment]
Yeah, that sounds fine. Happy to make changes that help portability. 

(The `autogen-version` script has become pretty messy, and we may reconsider how to do this at some point; but for now converting to Python sounds good, just no need to spend much time on making it pretty 😄 )
